// @sap/ux-specification/dist/src/apiTypes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FPMWriterMap = exports.CustomExtensionType = exports.CustomActionTargetControl = void 0;
const fe_fpm_writer_1 = require("@sap-ux/fe-fpm-writer");
Object.defineProperty(exports, "CustomActionTargetControl", { enumerable: true, get: function () { return fe_fpm_writer_1.TargetControl; } });
var CustomExtensionType;
(function (CustomExtensionType) {
    CustomExtensionType["CustomPage"] = "CustomPage";
    CustomExtensionType["CustomColumn"] = "CustomColumn";
    CustomExtensionType["CustomSection"] = "CustomSection";
    CustomExtensionType["ObjectPage"] = "ObjectPage";
    CustomExtensionType["CustomAction"] = "CustomAction";
    CustomExtensionType["CustomView"] = "CustomView";
    CustomExtensionType["ControllerExtension"] = "ControllerExtension";
})(CustomExtensionType = exports.CustomExtensionType || (exports.CustomExtensionType = {}));
exports.FPMWriterMap = new Map([
    [CustomExtensionType.CustomPage, fe_fpm_writer_1.generateCustomPage],
    [CustomExtensionType.CustomColumn, fe_fpm_writer_1.generateCustomColumn],
    [CustomExtensionType.CustomSection, fe_fpm_writer_1.generateCustomSection],
    [CustomExtensionType.ObjectPage, fe_fpm_writer_1.generateObjectPage],
    [CustomExtensionType.CustomAction, fe_fpm_writer_1.generateCustomAction],
    [CustomExtensionType.CustomView, fe_fpm_writer_1.generateCustomView],
    [CustomExtensionType.ControllerExtension, fe_fpm_writer_1.generateControllerExtension]
]);
